### PR TITLE
Standalone materialize method

### PR DESCRIPTION
**Pull request by dpeng817**
barebones standalone materialize method. Lmk if there are any test cases worth addressing here that I missed due to my lack of asset edge cases knowledge.

### CODE REVIEW

File: python_modules/dagster/dagster/core/asset_defs/materialize.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def materialize(
+    assets: Sequence[AssetsDefinition],
+    source_assets: Optional[Sequence[SourceAsset]] = None,
+    run_config: Any = None,
+    instance: Optional[DagsterInstance] = None,
+) -> ExecuteInProcessResult:
+    """
+    Executes an in-process run which materializes assets that fulfill selection.
```
Comment (clairelin135) [inline python_modules/dagster/dagster/core/asset_defs/materialize.py:22]: I think it would be good to mention in this docstring that this method will write asset outputs to the filesystem (via `fs_io_manager`).

**Review by clairelin135 — APPROVED** [review]
Looking good to me after we add the additional test cases owen mentioned

File: python_modules/dagster/dagster/core/asset_defs/materialize.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster/core/asset_defs/materialize.py:22]: I think also worth mentioning that the materializations will happen serially in a single thread.

**Comment by sryza** [comment]
`load_assets_from_package_module` returns a list of AssetsDefinitions and SourceAssets, but `materialize` in the form it's in here accepts these in separate lists. Which means that `materialize(load_assets_from_package_module(...))` won't always work. I wonder if we should have `materialize` accept a list that can have both types? There might be downsides I'm not thinking of though?

**Comment by dpeng817** [comment]
> `load_assets_from_package_module` returns a list of AssetsDefinitions and SourceAssets, but `materialize` in the form it's in here accepts these in separate lists. Which means that `materialize(load_assets_from_package_module(...))` won't always work. I wonder if we should have `materialize` accept a list that can have both types? There might be downsides I'm not thinking of though?

I think this is potentially problematic when you want to use assets as source assets but not materialize them? There's no way to do that if you put everything as one list right?

**Comment by sryza** [comment]
> I think this is potentially problematic when you want to use assets as source assets but not materialize them? There's no way to do that if you put everything as one list right?

`SourceAsset` objects can't be materialized no matter what though, right?
